lib/api.js

```diff
@@ -288,21 +325,115 @@ API.prototype.export = function(opts) {
 API.prototype.import = function(str, opts) {
   opts = opts || {};
 
-  var credentials;
+
+  // Json input?
   try {
+    var credentials;
     if (opts.compressed) {
       credentials = Credentials.importCompressed(str, opts.password);
-      // HACK: simulate incomplete credentials
-      delete credentials.m;
     } else {
       credentials = Credentials.fromObj(JSON.parse(str));
     }
+    this.credentials = credentials;
   } catch (ex) {
-    throw new Error('Error importing from source:' + ex);
+    throw Errors.INVALID_BACKUP;
   }
-  this.credentials = credentials;
 };
 
+
+API.prototype.importFromMnemonic = function(words, opts, cb) {
+  log.debug('Importing from 12 Words');
+
+  var key;
+  try {
+    key = Credentials.keyFromMnemonic(words, opts.passphrase, opts.network);
+  } catch (e) {
+    log.info('Mnemonic error:', e);
+    return cb(Errors.INVALID_BACKUP);
+  };
+
+  return this.importFromExtendedPrivateKey(key, cb);
+};
+
+
+API.prototype.importFromExtendedPrivateKey = function(xPrivKey, cb) {
+  var self = this;
+
+  log.debug('Importing from Extended Private Key');
+  this.credentials = Credentials.fromExtendedPrivateKey(xPrivKey);
+
+  // First option, grab wallet info from BWS.
+  self.openWallet(function(err, ret) {
+    // it worked?
+    // TODO download encrypted Wallet key ...
+    if (ret) return cb(null, ret);
+
+    log.info('Copayer not found, trying to add access');
+
+    //Second option, lets try to add an access
+    self.addAccess({}, function(err) {
+
+      // it worked?
+      if (!err) 
+        self.openWallet(cb);
```

Comment:
Isn't this too optimistic? What if the wallet cannot be opened for some other reason? Won't this cause the app to fill up the access key list?